packages/state-transition/src/cache/epochCache.ts

```diff
@@ -918,12 +918,20 @@ export class EpochCache {
    * balances are sampled to adjust the probability of the next selection (32 per epoch on average). So to invalidate
    * the prediction the effective of one of those 32 samples should change and change the random_byte inequality.
    */
-  getBeaconProposersNextEpoch(): ValidatorIndex[] {
+  async getBeaconProposersNextEpoch(): Promise<ValidatorIndex[]> {
     if (!this.proposersNextEpoch.computed) {
+      const shuffling = await this.shufflingCache?.get(this.nextEpoch, this.nextDecisionRoot);
```

Comment:
I think we need to handle the case where this is invoked concurrently, otherwise we end up running shuffling calculation multiple times as `this.proposersNextEpoch.computed` update will be delayed until after Promise resolves

Reply:
That's actually handled in the shuffling cache. Async `get` returns the same promise from multiple calls and will resolve all at the same time. So there will be multiple settings of the resolution in the EpochCache but only one calculation. As a note this will still throw if there was no request to calculate the shuffling in an epoch transition.